components/raftstore/src/coprocessor/dispatcher.rs

```diff
@@ -635,6 +635,26 @@ impl<E: KvEngine> CoprocessorHost<E> {
         );
     }
 
+    /// `pre_commit` is called we we want to persist data or meta for a region.
+    /// For example, in `finish_for` and `commit`,
+    /// we will separately call `pre_commit` with is_finished = true/false.
+    /// By returning false, we reject this persistence.
+    pub fn pre_commit(
```

Comment:
Better give it a better name without leaking the implementation details. It will be changed in v2.